src/lib.rs

```diff
@@ -187,6 +187,13 @@ impl Vfs {
         self.roots.path(root).to_path_buf()
     }
 
+    pub fn path2root(&self, path: &Path) -> Option<VfsRoot> {
+        self.roots.iter().find(|vfs_root| {
+            let root_path = self.roots.path(*vfs_root);
+            path.starts_with(root_path)
+        })
+    }
```

Comment:
This should use https://github.com/rust-analyzer/ra_vfs/blob/master/src/roots.rs#L81-L86 I think, to propertly account for nested roots

Reply:
I tried using find initially, but it doesn't seem to find a root at all

Reply:
Ahh, because the default filter ignores `target/` the path isn't counted as being part of the VfsRoot